include/solvers/mpm_base.tcc

```diff
@@ -937,6 +940,62 @@ void mpm::MPMBase<Tdim>::nodal_frictional_constraints(
   }
 }
 
+// Nodal absorbing constraints
+template <unsigned Tdim>
+void mpm::MPMBase<Tdim>::nodal_absorbing_constraints(
+    const Json& mesh_props, const std::shared_ptr<mpm::IOMesh<Tdim>>& mesh_io) {
+  try {
+    // Read and assign absorbing constraints
+    if (mesh_props.find("boundary_conditions") != mesh_props.end() &&
+        mesh_props["boundary_conditions"].find("absorbing_constraints") !=
+            mesh_props["boundary_conditions"].end()) {
+      // Iterate over velocity constraints
+      for (const auto& constraints :
+           mesh_props["boundary_conditions"]["absorbing_constraints"]) {
+        // Absorbing constraints are specified in a file
+        if (constraints.find("file") != constraints.end()) {
+          std::string absorbing_constraints_file =
+              constraints.at("file").template get<std::string>();
+          bool absorbing_constraints =
+              constraints_->assign_nodal_absorbing_constraints(
+                  mesh_io->read_absorbing_constraints(
+                      io_->file_name(absorbing_constraints_file)));
+          if (!absorbing_constraints)
+            throw std::runtime_error(
+                "Absorbing constraints are not properly assigned");
+
+        } else {
```

Comment:
This is not needed, let's not read it from a file

Reply:
outdated - resolved